scripts/smoke.sh

```
@@ -0,0 +1,404 @@
+  log "Starting ${name}: image=${image} port=${port} path=${path}"
+  if ! cid=$(docker run -d --name "$container" -p "${port}:3000" $envflags $DOCKER_FLAGS "$image" 2>"$TMP_DIR/${container}.err"); then
+    # If the image tag is :ci-smoke and it fails, try falling back to :latest
+    base="${image%:*}"
+    tag="${image##*:}"
+    if [[ "$tag" == "$image" ]]; then
+      tag=""
+    fi
+    if [[ "$tag" == "ci-smoke" ]]; then
+      alt_image="${base}:latest"
+      log_warn "Failed to start ${name} with ${image}, retrying with ${alt_image} ..."
+      if ! cid=$(docker run -d --name "$container" -p "${port}:3000" $envflags $DOCKER_FLAGS "$alt_image" 2>"$TMP_DIR/${container}.err"); then
```

Shellcheck (0.10.0)

[warning] cid appears unused. Verify use (or export if used externally).

(SC2034)